estate/models/estate_property.py

```diff
@@ -0,0 +1,123 @@
+from odoo import models, fields, api
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_compare, float_is_zero
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+    _order = "id desc"
+    
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)', 'Expected price must be strictly positive'),
+        ('check_selling_price', 'CHECK(selling_price >= 0)', 'Selling price must be positive'),
+    ]
+
+    # Basic Information
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, default=lambda self: fields.Date.add(fields.Date.today(), months=2))
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ], default='new')
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    # Pricing
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    best_price = fields.Float(compute='_compute_best_price', string='Best Offer')
+    
+    # Property Details
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    total_area = fields.Integer(compute='_compute_total_area', string='Total Area')
+    
+    # Features
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        selection=[
+            ('north', 'North'),
+            ('south', 'South'),
+            ('east', 'East'),
+            ('west', 'West'),
+        ]
+    )
+    
+    # People
+    salesperson_id = fields.Many2one('res.users', string='Salesperson', default=lambda self: self.env.user)
+    buyer_id = fields.Many2one('res.partner', string='Buyer', copy=False)
+    
+    # Tags
+    tag_ids = fields.Many2many('estate.property.tag', string='Tags')
+    
+    # Offers
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offers')
+    
+    # Computed Methods
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
```

Comment:
```suggestion
        for property in self:
```
We tend to use a variable that reflects the model were are in